plugins/woocommerce/src/Internal/Admin/Logging/PageController.php

```diff
@@ -0,0 +1,316 @@
+				),
+				'source'  => FILTER_SANITIZE_STRING,
+			),
+			false
+		);
+		$params   = wp_parse_args( $params, $defaults );
+
+		return $params;
+	}
+
+	/**
+	 * Get and cache an instance of the list table.
+	 *
+	 * @return ListTable
+	 */
+	private function get_list_table(): ListTable {
+		if ( $this->list_table instanceof ListTable ) {
+			return $this->list_table;
+		}
+
+		$this->list_table = new ListTable( $this->file_controller, $this );
+
+		return $this->list_table;
+	}
+
+	/**
+	 * Register screen options for the logging views.
+	 *
+	 * @return void
+	 */
+	private function setup_screen_options(): void {
+		$params = $this->get_query_params();
+
+		if ( 'list_files' === $params['view'] ) {
+			// Ensure list table columns are initialized early enough to enable column hiding.
+			$this->get_list_table()->prepare_column_headers();
+
+			add_screen_option(
+				'per_page',
+				array(
+					'default' => 20,
+					'option'  => ListTable::PER_PAGE_USER_OPTION_KEY,
+				)
+			);
+		}
+	}
+
+	/**
+	 * Process bulk actions initiated from the log file list table.
+	 *
+	 * @return void
+	 */
+	private function handle_list_table_bulk_actions(): void {
+		$action = $this->get_list_table()->current_action();
+
+		// phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
+		$request_uri = isset( $_SERVER['REQUEST_URI'] ) ? wp_unslash( $_SERVER['REQUEST_URI'] ) : $this->get_logs_tab_url();
+
+		if ( $action ) {
+			check_admin_referer( 'bulk-log-files' );
+
+			$sendback = remove_query_arg( array( 'deleted' ), wp_get_referer() );
+			$files    = filter_input( INPUT_GET, 'file', FILTER_SANITIZE_STRING, FILTER_REQUIRE_ARRAY );
+
+			if ( ! is_array( $files ) || count( $files ) < 1 ) {
+				wp_safe_redirect( $sendback );
+			}
+
+			switch ( $action ) {
+				case 'delete':
+					$deleted  = $this->file_controller->delete_files( $files );
+					$sendback = add_query_arg( 'deleted', $deleted, $sendback );
+					break;
+			}
+
+			$sendback = remove_query_arg( array( 'action', 'action2' ), $sendback );
+
+			wp_safe_redirect( $sendback );
+			exit;
+		} elseif ( ! empty( $_REQUEST['_wp_http_referer'] ) ) {
+			$removable_args = array( '_wp_http_referer', '_wpnonce', 'action', 'action2', 'filter_action' );
+			wp_safe_redirect( remove_query_arg( $removable_args, $request_uri ) );
+			exit;
+		}
+
+		$deleted = filter_input( INPUT_GET, 'deleted', FILTER_VALIDATE_INT );
+
+		if ( is_numeric( $deleted ) ) {
+			add_action(
+				'admin_notices',
+				function() use ( $deleted ) {
+					?>
+					<div class="notice notice-info is-dismissible">
+						<p>
+							<?php
+							printf(
+								// translators: %s is a number of files.
+								esc_html( _n( '%s log file deleted.', '%s log files deleted.', $deleted, 'woocommerce' ) ),
+								esc_html( number_format_i18n( $deleted ) )
+							);
+							?>
+						</p>
+					</div>
+					<?php
+				}
+			);
+		}
+
+		$_SERVER['REQUEST_URI'] = remove_query_arg( array( 'deleted' ), $request_uri );
```

Comment:
It's usually better not to modify super-globals like this one. Is there a better way to do this (I was actually a little unsure why it's needed)?

Reply:
I based this (rather ugly) handle routine off the one in WP Core, and it modifies the super global here:
https://github.com/WordPress/WordPress/blob/e5d1fca19850a1d47b4f62303a24aca8797d3efb/wp-admin/edit.php#L483

I assumed that it kept the `deleted` param from staying a part of the URL, but I just tried the bulk edit with that line commented out, and the param was still removed anyway... 🤔

Reply:
Removed in ﻿73924c6107